cumulative/cumulative.go

```diff
@@ -97,7 +111,51 @@ func (dc *DeltaCalculator) CountMetric(name string, attributes map[string]interf
 		}
 	}
 	if !ok || timestampsOrdered {
-		dc.datapoints[id] = lastValue{value: val, when: now}
+		dc.countDatapoints[id] = lastCountValue{value: val, when: now}
+	}
+	return
+}
+
+// SummaryMetric creates a summary metric from the difference between the values and
+// timestamps of multiple calls.  If this is the first time the name/attributes
+// combination has been seen then the `valid` return value will be false.
+func (dc *DeltaCalculator) SummaryMetric(name string, attributes map[string]interface{}, val SummaryValue, now time.Time) (summary telemetry.Summary, valid bool) {
+	var attributesJSON []byte
+	if nil != attributes {
+		attributesJSON = internal.MarshalOrderedAttributes(attributes)
+	}
+	dc.lock.Lock()
+	defer dc.lock.Unlock()
+
+	if now.Sub(dc.lastClean) > dc.expirationCheckInterval {
+		cutoff := now.Add(-dc.expirationAge)
+		for k, v := range dc.summaryDatapoints {
+			if v.when.Before(cutoff) {
+				delete(dc.summaryDatapoints, k)
+			}
+		}
+		dc.lastClean = now
+	}
```

Comment:
What do you think about pulling this and the similar section from the Count above into a separate method. Then in that method clean both the summary and count datapoints?